### ISSUE TITLE
Cache not clearing after saving Commerce products

### ISSUE THREAD

**Issue by philipboomy** [putyourlightson/craft-blitz#474]
I am running into products not refreshing after clicking save.

1) Selecting some products in a "Product field" and clicking save
2) Go to frontend and check and page is still cached.
3) Change a text field and clicking save.
4) Go to frontend and check and page is still cached.
5) Edit the title field and click save.
6) Go to frontend and check and page is now refreshed.

I waited till queue was done running just in case. Didn't change anything.

Logging is on locally but nothing in the log.

Blitz 4.3.0 and Craft CMS 4.3.8.1

**Comment by bencroker** [comment]
Are you saying that the product page is not cleared, or an element that is _related to_ a product? Please provide more details so I can replicate locally.

**Comment by philipboomy** [comment]
Thanks for quick repsonse.

The product I am editing is not cleared. The product field is just one type of edit I am doing on the product. Another is the text field in step 3.

**Comment by bencroker** [comment]
Ok, can you please rephrase it to the simplest example which is not working as expected, with steps to replicate?

**Comment by philipboomy** [comment]
1) Edit a text field on a product
2) Save product
3) Visit product frontend
4) See that product is cached and the change in step 1 is not showing frontend.

**Comment by bencroker** [comment]
Thank you, I'll investigate that use-case today.

**Comment by bencroker** [comment]
I believe this is due to products not auto-saving drafts, which I've fixed in https://github.com/putyourlightson/craft-blitz/commit/3473b56eca362c5cd77ca8b750d452d0813843e1 for the next release.

**Comment by bencroker** [comment]
On second look, this method looks like it's too eager – fields are _always_ marked as dirty when saved in the CP, regardless of whether they have changed or not. I'll need to look into whether something has changed in recent Craft versions and otherwise come up with a workaround.

**Comment by ZapaNOR** [comment]
I can confirm blitz does not seem to clear cache on normal entries too:

1. Edit redactor field on entry
2. Save entry, wait for all queue jobs to finish (did not see any clearing blitz cache jobs).
3. Refresh frontend, changes not out.
4. Clear cache in utilities, changes appear.

We have this problem on two separate sites now after updating Blitz to latest.

Edit: We use "Clear the cache and regenerate in a queue job" as refresh mode.

**Comment by bencroker** [comment]
@Mathias-Syversen did you update Craft as well as Blitz? Just trying to figure out which update might have caused this change in behaviour.

**Comment by bencroker** [comment]
@Mathias-Syversen I just ran a local test and modifying an entry _does_ refresh the cache as expected. Can you please create a new issue with more details so as to not confuse the issue we're working through here with Commerce products?

**Comment by bencroker** [comment]
It looks like elements that do not support drafts, which includes Commerce products, were not triggering cache refreshes as of version 4.3.0. I've fixed in https://github.com/putyourlightson/craft-blitz/commit/3473b56eca362c5cd77ca8b750d452d0813843e1 and released in version 4.3.1. It would be great if you could confirm that this fixes your issue @philipboomy.

**Comment by philipboomy** [comment]
@bencroker I can confirm it is fixed. Cache is clearing after saving a product.
